### DISCUSSION TITLE
Haunches on steel members in SAF

### DISCUSSION THREAD

**Discussion by sonkaste17** [StructuralAnalysisFormat/StructuralAnalysisFormat-Doc#218] [Ideas]
Haunches on steel members in SAF. How to define haunches in SAF? They are missed at the moment.

![image](https://user-images.githubusercontent.com/81751485/187197202-976e4922-0b18-447d-bb24-7ca4bcdbbdda.png)

**Comment by sonkaste17** [comment]
@msalak-scia please prepare a proposal for haunches, together with a parametric profile extension. 👍

**Comment by MartinSalak** [comment]
Hi all, you can see the proposal below. All comments are welcome!
\- will this object be easy to implement into your software?
\- how do you handle cross-sections like I+I var proposed below?
<br>

Modified object: **StructuralCurveMember**
New string: 
 | Property | Example | Definition |
| --- | --- | --- |
| Haunch | H1  | The reference to existing haunch in StructuralCurveMemeberHaunch |

New object: **StructuralCurveMemberHaunch**
StructuralCurveMemberHaunch creates a haunch on StructuralCurveMember. It assigns a new cross-section to the end of the beam and sets the distance at which it will gradually go back to the original cross-section of the beam.

| Property | Example | Definition |
| --- | --- | --- |
| Name | H1  | Human readable unique name of the haunch |
| Cross section | CS4 | The cross section from StructuralCrossSection used at the end(s) of the beam. Not all cross-sections can be used. They must be of similar type |
| Position | Begin,<br>End,<br> Both | Sets the position of the haunch |
| Alignment | Centre<br>Top<br>Bottom<br>Left<br>Right<br>Top left<br>Top right<br>Bottom left<br>Bottom right | Sets the alignment of the haunch to the original cross-section of the beam.<br>Centre – The center line remains straight. All surfaces can be symmetrically inclined to allow the change of the cross-section size.<br>Top - The top surface of the 1D member remains flat. The bottom surface is inclined to allow the change in height. The top surface is in direction of the vertical axis of CSS. <br>Bottom - The bottom surface of the 1D member remains flat. The bottom surface is in the opposite direction of the vertical axis of CSS.<br>Left - The left surface of the 1D member remains flat. The left surface is in direction of the horizontal axis of CSS.<br>Right - The right surface of the 1D member remains flat and horizontal. The right surface is in the opposite direction of the horizontal axis of CSS |
| Coordinate definition | Absolute<br>Relative | Specifies the definition of the length |
| Length x [m] | 0.2 | Defines the length of the haunch in relative or absolute coordinates. Length is always measured from the specified end(s). For the haunch on both ends, it is expected to be to the middle of the beam at maximum |
| Id | | |

![image](https://user-images.githubusercontent.com/89633938/212091332-aa96fb19-438a-40cd-b72e-89f86cdc1653.png)
<br>
<br>
**New parametric cross sections to Supported Shapes of Parametric Cross sections**
| Pattern | Shape | Parameters [mm] |
| --- | --- | --- |
| ![I welded param](https://user-images.githubusercontent.com/89633938/211852529-075d4fd7-ad43-4011-8263-8c5d3fe4269f.png) | I welded | Hw;B;s;t |
| ![I + I](https://user-images.githubusercontent.com/89633938/211852475-d9e9b2f3-ecfc-47a1-9318-2255d5f64c6f.png) | I + I var | I_section;va |
| ![I + 2 I](https://user-images.githubusercontent.com/89633938/211852416-72d652d8-eca7-4076-9762-d56a17e2a5e1.png) | I + 2 I var | I_section;va |
| ![I + PL](https://user-images.githubusercontent.com/89633938/211852489-a87e7983-e877-4332-9c4d-777d2d813412.png) | I + Pl | I_section;va;s |
| ![Iw + Iw](https://user-images.githubusercontent.com/89633938/211852577-ab2a0c35-849a-49dc-ad17-71022d1693ec.png) | Iw + Iw var | Hw;B;va;s;t |
| ![Iw + PL](https://user-images.githubusercontent.com/89633938/211852589-eca9158e-04ec-4b92-8c3a-039148915f30.png) | Iw + Pl | Hw;B;va;s;t |


**Notes**
* This object might seem similar to StructuralCurveMemberVarying. StructuralCurveMemberHaunch is only slightly changing cross section at the ends of the beam, whereas StructuralCurveMemberVarying is assigning a completely new cross section “over” the original to parts of the beam.
* Usually, this object is used to change the parameters of parametric cross-section towards the ends of the beam. It is not possible for all the cross-section. For example manufactured IPE200 with parts of IPE200 welded to it.  In this example, use IPE200 to the member and I + I with the same IPE200 as a parameter to the cross-section of the Haunch.
* From the end of the beam the parameters change linearly towards the length [x] where the cross-section becomes the original cross-section of the beam. For cross section like I + I the minimum value of their parameter is the thickness of their flange.
![image](https://user-images.githubusercontent.com/89633938/212100307-b0a72492-c486-4bb6-bff7-343ec5fc6660.png)
